### PR TITLE
Bug fixes for polls cms

### PR DESCRIPTION
**Pull request by collin**
- Don't allow setting published/end dates on new polls.
  This allows for a bizarre edge case where you might `POST { activationDate: null, ... }`
  to sodahead. They do not like this for new polls. Hiding the date fields until the poll
  exsists mitigates the issue.

- Fix the form validation, there was a flaw in the date validation logic that left
  `pollForm.$valid` as `undefined`, no bueno.

@MelizzaP @kand @spra85

### CODE REVIEW

**Comment by spra85** [comment]
:+1:

File: app/components/polls/polls-edit/polls-edit-directive.js [hunk 1]
```diff
@@ -54,18 +54,24 @@ angular.module('polls.edit.directive', [
         // in such a way that modal-on-close="validatePublication()"
         // fires before the scope model data has changed.
         $timeout(function () {
+          var pollForm = $scope.pollForm;
           var published = $scope.model.published;
           var endDate = $scope.model.end_date;
-          var publishedField = $scope.pollForm.published;
-          var endDateField = $scope.pollForm.endDate;
+          var publishedField = pollForm.published;
```
Comment (kand) [inline app/components/polls/polls-edit/polls-edit-directive.js]: Why the removal of `$scope` here?

**Comment by MelizzaP** [comment]
:+1:

**Comment by kand** [comment]
Otherwise, :+1:

File: app/components/polls/polls-edit/polls-edit-directive.js (hunk 1 above)
Comment (collin) [inline app/components/polls/polls-edit/polls-edit-directive.js]: It was an artifact of some debugging/changes to the way I was setting validations. Restored it.

**Pull request merged**